physionet-django/console/views.py

```diff
@@ -2985,41 +2985,88 @@ def event_management(request, event_slug):
 
     # handle the add dataset form(s)
     if request.method == "POST":
-        if 'add-event-dataset' in request.POST.keys():
+        if "add-event-dataset" in request.POST.keys():
             event_dataset_form = EventDatasetForm(request.POST)
             if event_dataset_form.is_valid():
-                if selected_event.datasets.filter(
-                        dataset=event_dataset_form.cleaned_data['dataset'],
-                        access_type=event_dataset_form.cleaned_data['access_type'],
-                        is_active=True).count() == 0:
+                active_datasets = selected_event.datasets.filter(
+                    dataset=event_dataset_form.cleaned_data["dataset"],
+                    access_type=event_dataset_form.cleaned_data["access_type"],
+                    is_active=True)
+                if len(active_datasets) == 0:
```

Comment:
Here it actually makes sense to use `.count()` rather than `len`. `len` makes sense when you already have the data loaded from the database and in your program's memory since it just iterates over the Python iterable to count the items. `.count()` fetches the length of the collection from the database, so if you already have the data on hand it's simpler to just count it locally rather than sending a request to the database.

In this case it's better to go for:

```suggestion
                if active_datasets.count() == 0:
```

Reply:
Thanks - I had the notion that the line about defining the filter leads to a query, but it actually does not trigger a query, and the query is only triggered when the variable is evaluated.